### DISCUSSION TITLE
how do I register my own theme?

### DISCUSSION THREAD

**Discussion by mep-bsw** [CodinGame/monaco-vscode-api#407] [Q&A]
Im trying to use a vs-code compatible theme, but dont know an exact way to get it working.
Im using the theme-service-override, but havent found a way to get it to accept my theme. are there any examples out there showing the process?

**Comment by CGNonofr** [comment]
VSCode extensions are providing themes, so you need one (either using the vsix loader to load the vsix directly, or using the `registerExtension` by hands)

**Comment by cemalgnlts** [comment]
I can add my own theme this way:
https://github.com/cemalgnlts/jside/tree/main/src/extensions/one-dark-theme

**Comment by mep-bsw** [comment]
can i use the filesystemoverride to get registerExtension working in a browser-environment?

i tried registering all the files mentioned in the manifest over the filesystemprovider like this:
```ts
    fileSystemProvider.registerFile(new RegisteredMemoryFile(Uri.file('/workspace/test'), ""));
    fileSystemProvider.registerFile(new RegisteredMemoryFile(Uri.file('/themes/pat-commands-light.json'), JSON.stringify(patCommandsLight)));
    registerFileSystemOverlay(1, fileSystemProvider);
    fileSystemProvider.registerFile(new RegisteredMemoryFile(Uri.file('/themes/pat-commands-light.json'), JSON.stringify(patCommandsLight)));
    registerExtension(manifest)
```
to no avail

**Reply by CGNonofr** [reply]
I guess it's possible to use a filesystem override to register an extension, but I'm not sure to see the point

What it the problem with the example @cemalgnlts provided?

**Reply by mep-bsw** [reply]
its very much much a hasle with webpack to get all the files provided. for now i have just switched to using a monaco-theme together with a default monaco-tokensprovider to achieve my goal of meaningful tokencolors. i need my own own tokenization in order for my ls and client to have the same tokens/save on work and not maintain multible grammars.
